src/highlight/HighlightAnnotations.tsx

```diff
@@ -80,11 +84,28 @@ const HighlightAnnotations = (props: Props): JSX.Element => {
         createHighlight({ ...staged, message });
 
         if (isPromoting) {
+            setMode(Mode.NONE);
             setIsPromoting(false);
         }
     };
 
     const handlePromote = (): void => {
+        if (!selection) {
+            return;
+        }
+
+        const { location, pageRect, rects } = selection;
+
+        setMode(Mode.HIGHLIGHT);
+        setStaged({
+            location,
+            shapes: rects.map(rect => ({
+                ...getShapeRelativeToPage(rect, pageRect),
+                type: 'rect',
+            })),
+        });
+        setStatus(CreatorStatus.staged);
+
         setIsPromoting(true);
```

Comment:
We're now firing six actions upon promotion, which seems excessive. Have we looked at handling some of these actions across reducers? For example, if we see `mode` set to `Mode.HIGHLIGHT`, can the promotion reducer reset its values transparently?